File: serp_llm/src/core/domain/entities/base_entity.py
# Author: QuanTuanHuy
# Description: Part of Serp Project - Base Entity

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field, ConfigDict


class BaseEntity(BaseModel):
    """Base entity with common fields"""
    
    model_config = ConfigDict(
        from_attributes=True,
        use_enum_values=True,
        populate_by_name=True,
    )
    
    id: Optional[int] = Field(default=None, description="Entity ID")
    created_at: Optional[datetime] = Field(default=None, description="Creation timestamp")
    updated_at: Optional[datetime] = Field(default=None, description="Update timestamp")

File: pm_core/src/main/java/serp/project/pmcore/domain/entity/BaseEntity.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public abstract class BaseEntity {
    private Long id;
    private Long createdAt;
    private Long createdBy;
    private Long updatedAt;
    private Long updatedBy;

    public void applyCreate(Long createdBy, Long createdAt) {
        this.createdBy = createdBy;
        this.createdAt = createdAt;
        this.updatedAt = createdAt;
        this.updatedBy = createdBy;
    }

    public void applyUpdate(Long updatedBy, Long updatedAt) {
        this.updatedBy = updatedBy;
        this.updatedAt = updatedAt;
    }
}
